A MissingSectionHeader error now also reports the cached file

reinout opened the pull request:
Fixes #245 mostly.

The error now prints the cached location of the downloaded file, making it much easier to clean up downloads that went wrong.

reinout commented:
Ok, waiting for someone to take a seond look :-)

reinout commented on src/zc/buildout/buildout.py:
```diff
@@ -1570,9 +1570,10 @@ def _open(base, filename, seen, dl_options, override, downloaded):
-        path, is_temp = download(filename)
-        fp = open(path)
+        cached_filename, is_temp = download(filename)
+        fp = open(cached_filename)
```
I basically just renamed `path` to the more explicit `cached_filename`.
This variable is then used to print the cache's filename in addition to the original one when there's an exception.

mgedmin commented:
Very nice!

Patch looks good to me.

I assume that, if an error happens, `zc.buildout.configparser.parse` will raise and therefore the `if is_temp: os.unlink(cached_filename)` branch won't be executed?

reinout commented:
@mgedmin: yes, the error is raised and thus prevents the removal of the temp file.

That *does* mean I have to change the wording a little bit. I now say "http://url (cached at xyz)", but that "cached" might also mean "this is the temp file where I downloaded it to.

What about "downloaded as xyz" instead of "cached at xyz"? That would cover both cached files and temp files.

reinout commented:
@mgedmin: I've changed `cached_filename` to `downloaded_filename`. And I've changed the wording of the error message. This should cover both use cases, right?

mgedmin commented:
Yes, looks good.

:shipit:
